README.md

```diff
@@ -1,190 +1,97 @@
 
-```scss
-$o-header-primary-left-width
-$o-header-secondary-left-width
-$o-header-primary-featured-width
+```js
+require('o-header');
+document.addEventListener("DOMContentLoaded", function() {
+    document.dispatchEvent(new CustomEvent('o.DOMContentLoaded'));
+});
 ```
 
-You may specify specific widths for different layouts, [based on o-grid](https://github.com/Financial-Times/o-grid):
+Additionally, a second paramater can be passed to the header constructor or to the `.init()` function with a config object that has the following options:
 
-```scss
-// always 254px wide
-$o-header-primary-left-width: 254px;
+* *headerClassName*: Class name set to the root element of your head. (Default: `o-header`)
+* *editionswitchClassName*: Class name set to the root element of the edition switch. (Default: `o-header__edition-switch`)
 
-// 254px wide by default, and 303px wide at the medium layout and up
-$o-header-primary-left-width: (default: 254px, M:  303px);
-```
+### Sass
 
-## JavaScript instantiation
+Currently, only non silent mode is supported.
 
-An __o-header__ object must be constructed for every `<header>` you have on your page that uses this module.
+### Markup
 
-```javascript
-var oHeader = require('o-header');
-var header = document.querySelector('.o-header');
-var ftHeader = new oHeader(header);
-```
+_There are intentionally no classes to switch between logged in and out as we don't want to do that in the client side. This is left up to the product._
 
-Alternatively, a `o.DOMContentLoaded` event can be dispatched on the `document` to auto-construct an __o-header__ object for each element with a `data-o-component="o-header"` attribute:
+Some elements inside the header require specific data attributes so the JavaScript can add some behaviour correctly. These are:
 
-```javascript
-require('o-header');
-document.addEventListener("DOMContentLoaded", function() {
-	document.dispatchEvent(new CustomEvent('o.DOMContentLoaded'));
-});
-```
+* data-o-header-search: Applied to the search `<form>`. Inside, it also requires an `<input>` element
+* data-o-header-togglable: Applied to any toggle in the header
+* data-o-header-togglable-nav: Applied to the `<button>` or other kind of element that will toggle the meganav
+* data-o-header-togglable-search: Applied to the `<button>` or other kind of element that will toggle the search input
+* data-o-header-suggestions: Applied to the root div of the bottom row of the header, as that's the container of the suggestions that are added via JS
+* data-o-header-edition-switch-button: Applied to the `<button>` or other kind of element that will toggle the edition switch
+* data-o-header-selectable: Applied to the `<button>` or other kind of element in the meganav section in the mobile view that will select it
 
-## Upgrading from 3.x.x to 4.x.x
```

Comment:
it is a complete overhaul. Do you think it's worth having a section to mention that? Happy to add it